File: backend/BusinessObjects/Models/Milestone.cs
using BusinessObjects.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace BusinessObjects.Models
{
    public class Milestone
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("milestone_id")]
        public long MilestoneId { get; set; }
        [Column("milestone_name")]
        public string MilestoneName { get; set; } = string.Empty;
        [Column("project_id")]
        public long ProjectId { get; set; }
        [Column("deadline_date")]
        public DateTime DeadlineDate { get; set; }
        [Column("finish_date")]
        public DateTime? FinishDate { get; set; }
        [Column("status")]
        public MilestoneStatus Status { get; set; }
        [Column("milestone_description")]
        public string MilestoneDescription { get; set; } = string.Empty;
        [Column("pay_amount")]
        public decimal PayAmount { get; set; }
        // Navigation property
        [ForeignKey("ProjectId")]
        [JsonIgnore]
        public virtual Project Project { get; set; } = null!;
    }
}

File: frontend/src/types/milestone.ts
export enum MilestoneStatus {
  NOT_STARTED = 0,
  IN_PROGRESS = 1,
  SUBMITTED = 2,
  COMPLETED = 3,
  CANCELLED = 4,
}

export type CreateMilestoneDTO = {
  milestoneName: string;
  description: string;
  amount: number;
  deadline: string;
};

export interface Milestone {
  milestoneId: string;
  milestoneName: string;
  milestoneDescription: string;
  amount: number;
  deadlineDate: string;
  payAmount: number;
  projectId: string;
  status: MilestoneStatus;
}
